src/osd/PGLog.cc

```diff
@@ -865,40 +872,39 @@ void PGLog::_write_log_and_missing(
   eversion_t write_from_dups,
   bool *may_include_deletes_in_missing_dirty, // in/out param
   set<string> *log_keys_debug,
+  set<string> *to_remove,
+  set<std::pair<string, string>> *to_rmkeyrange,
```

Comment:
Let's distinguish the parameter names which are expected to be handled by the newly introduced log interfaces, e.g.:
* km -> log_to_setkeys
* to_remove -> log_to_rmkeys
* to_rmkeyrange -> log_to_rmkeyranges

This is because any mismatch (e.g. store through log interfaces but load from omap interfaces) will lead to errors.

Reply:
Fixed.

Reply:
If `_write_log_and_missing()` is going to be reused, can you rename the parameter `km` to `log_to_setkeys` for the same reason ?

A rough look at `FuturizedShardStoreLogReader::read()`, seems the special keys (such as "can_rollback_to", etc.) are read through `t.log_setkeys()` correctly.

Reply:
Done.